storcli.py

```diff
@@ -57,8 +64,178 @@ def main(args):
     except KeyError:
         pass
 
+    parsers = {
+        'cv': handle_cachevault,
+        'bbu': handle_bbu
+    }
+
+    for parser, parser_func in parsers.items():
+        data = get_storcli_json('/cALL/' + parser + ' show all J')
```

Comment:
This will be run even if args.no_detailed_bbu is set.  I believe there is sufficient info in the main "all" output for users who don't want detailed info. Examples:

```
                "BBU_Info" : [
                        {
                                "Model" : "iBBU",
                                "State" : "Dgd (Needs Attention)",
                                "RetentionTime" : "48 hours +",
                                "Temp" : "29C",
                                "Mode" : "-",
                                "MfgDate" : "2014/02/10",
                                "Next Learn" : "2019/06/27  01:33:42"
                        }
                ]

```

```
                "Cachevault_Info" : [
                        {
                                "Model" : "CVPM02",
                                "State" : "Optimal",
                                "Temp" : "30C",
                                "Mode" : "-",
                                "MfgDate" : "2014/05/30"
                        }
                ]
```

Reply:
Yes, that's what I meant in the first bullet of [my other comment](https://github.com/prometheus-community/node-exporter-textfile-collector-scripts/issues/27#issuecomment-568042283), I think it's more clear to have a single branch dealing with the specifics (collaterally covering the "general overview" from "/call show all" that you printed above) than having bbu/cv metrics coming from both codepaths.

The only advantage I see in going back to collecting those 2 metrics is to avoid two more system calls, I just took that as neglictible. In that sense, the option "detailed_bbu" also changed to "no_detailed_bbu". From my experience, the users generally won't mind having more metrics on a new version.